### PR TITLE
refactor: change deploy script output file format

### PR DESCRIPTION
**Pull request by dohaki**
This changes the deploy script to be compatible with the newly created https://github.com/bosonprotocol/instances repository. If we deploy contracts from this repository then the contract addresses now get written to `<ROOT>/addresses/<CHAIN_ID>-<ENV>.json` which are not checked in to git. The outputted file can then be consumed by a script of the https://github.com/bosonprotocol/instances repository to easily update addresses.

This also removes the `contracts-<ENV>.json` files, as they are now maintained in a separate repository.

The change allows the contract deployments to be decoupled from the protocol release and enables a single source of truth for official contract addresses and configurations.

### CODE REVIEW

File: scripts/utils.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+export const addressesDirPath = __dirname + `/../addresses`;
+
+const availableEnvironments = [
+  'testing',
+  'staging',
+  'production',
+  'hardhat',
+  '',
```
Comment (zajck) [inline scripts/utils.ts:8]: Is there any reason, why allow the empty string?

**Review by zajck — CHANGES_REQUESTED** [review]

File: scripts/utils.ts (hunk 1 above)
Comment (dohaki) [reply scripts/utils.ts:8]: My initial idea was to keep `--env` optional as it was before. So the default argument of `--env` would be `""` and the output `./addresses/<CHAIN_ID>.json`.

Comment (zajck) [reply scripts/utils.ts:8]: Makes sense to me.

**Review by zajck — APPROVED** [review]

**Review by cliffhall — APPROVED** [review]
LGTM! 👍

**Pull request merged**